### PR TITLE
add riscv feature by default for benchmarks

### PR DESCRIPTION
**Pull request by mordamax**

### CODE REVIEW

**Comment by athei** [comment]
Did you test that? I don't think all runtimes have the `riscv` feature. Only the kitchensink runtime so far.

**Comment by mordamax** [comment]
> Did you test that? I don't think all binaries have the `riscv` feature? I am not sure.

Do you mean rustc-rv32e-toolchain ?
i was testing it meanwhile - https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/7431951
looks like it doesn't compile yes, but not sure what's the issue, I asked @xermicus in dm, but may be you could advice

**Comment by athei** [comment]
No. What I mean is that if you supply `--features riscv` the crate you are compiling needs to have this feature. I don't think its the case for all the invocations you specified here.

**Comment by mordamax** [comment]
> No. What I mean is that if you supply `--features riscv` the crate you are compiling needs to have this feature. I don't think it's the case for all the invocations you specified here.

Then I was probably confused with this statement https://github.com/paritytech/polkadot-sdk/issues/5714#issuecomment-2352187172

I thought you can provide feature and if it doesn't exist anywhere it will skip or something. 
Could you elaborate then what did you mean by `Just always enabling the riscv feature when benchmarking is good enough` ?

**Comment by athei** [comment]
Unfortunately, that is not how cargo works. The crate you are building has to have the feature or it will error out. This means **only** add the feature when building a node that has the feature. Only add the feature when building the `staging-node-cli` binary.

We can add the feature to other nodes, too. But I don't understand the script here. How many different nodes it is building here for?

**Comment by athei** [comment]
The first point is a bit wasteful. The second will not work as it wouldn't pass the feature into the wasm build. For that it has to be defined on the runtime and that one passes it down.

I would say we just add the feature to the nodes (last point). But I have to understand which nodes are actually build by this script.

**Comment by athei** [comment]
Okay yeah I checked the script. I will add the features the other nodes.

File: commands/bench/lib/bench-all-cumulus.sh [hunk 1]
```diff
@@ -74,7 +74,7 @@ run_cumulus_bench() {
 
 
 echo "[+] Compiling benchmarks..."
-cargo build --profile $profile --locked --features=runtime-benchmarks -p polkadot-parachain-bin
+cargo build --profile $profile --locked --features=runtime-benchmarks,riscv -p polkadot-parachain-bin
```
Comment (athei) [inline commands/bench/lib/bench-all-cumulus.sh:77]: I will add the feature to this node. We will need this once we add pallet_revive to westend anyways.

File: commands/bench/lib/bench-all-polkadot.sh [hunk 2]
```diff
@@ -12,7 +12,7 @@ runtime="${out:-""}"
 export RUST_LOG="${RUST_LOG:-error}"
 
 echo "[+] Compiling benchmarks..."
-cargo build --profile $profile --locked --features=runtime-benchmarks -p polkadot
+cargo build --profile $profile --locked --features=runtime-benchmarks,riscv -p polkadot
```
Comment (athei) [inline commands/bench/lib/bench-all-polkadot.sh:15]: The polkadot binary will never have this feature. Just revert this change.

```suggestion
cargo build --profile $profile --locked --features=runtime-benchmarks -p polkadot
```

File: commands/bench/lib/bench-all-trappist.sh [hunk 3]
```diff
@@ -14,7 +14,7 @@ chain="${runtime}-dev"
 export RUST_LOG="${RUST_LOG:-error}"
 
 echo "[+] Compiling benchmarks..."
-cargo build --profile $profile --locked --features=runtime-benchmarks
+cargo build --profile $profile --locked --features=runtime-benchmarks,riscv
```
Comment (athei) [inline commands/bench/lib/bench-all-trappist.sh:17]: I have no idea what this is building as no `-p` is supplied. But just remove this as whatever trappist is has no riscv feature probably.

File: commands/bench/lib/bench-pallet.sh [hunk 4]
```diff
@@ -74,7 +74,7 @@ bench_pallet() {
 
       args=(
         --bin=polkadot
-        --features=runtime-benchmarks
+        --features=runtime-benchmarks,riscv
```
Comment (athei) [inline commands/bench/lib/bench-pallet.sh:77]: Remove. Polkadot will not have this feature.

File: commands/bench/lib/bench-pallet.sh [hunk 5]
```diff
@@ -144,7 +144,7 @@ bench_pallet() {
       local weights_dir="$output_path/runtime/$runtime/src/weights"
 
       args=(
-        --features=runtime-benchmarks
+        --features=runtime-benchmarks,riscv
```
Comment (athei) [inline commands/bench/lib/bench-pallet.sh:147]: Remove

**Comment by athei** [comment]
https://github.com/paritytech/polkadot-sdk/pull/5832

**Comment by athei** [comment]
Changed my mind. Let's **only** add it for the kitchensink node.

**Comment by athei** [comment]
The title of this PR is wrong now, right? Because it parses additional features from the cmd instead of adding riscv?

**Comment by athei** [comment]
Is this now adding riscv by default or does it allow to specify features?

**Comment by mordamax** [comment]
> Is this now adding riscv by default or does it allow to specify features?

both ) it adds to dev/kitchensink, but also allows you to provide it --feature=blabla

**Review by athei — APPROVED** [review]

**Review by xermicus — APPROVED** [review]

**Comment by athei** [comment]
Do we even need that since this is the deprecated approach and we can do everything with `/cmd`? Asking because eventually we will remove the `riscv` feature once we have an upstream toolchain.

**Comment by mordamax** [comment]
> Do we even need that since this is the deprecated approach and we can do everything with `/cmd`? Asking because eventually we will remove the `riscv` feature once we have an upstream toolchain.

good point. I think it's fine, as I hope we anyway deprecate this bot as soon as we confirm /cmd works well for all use-cases, and we just archive this repo

**Review by mutantcornholio — APPROVED** [review]
